### DISCUSSION TITLE
Set default connection settings for nrepl

### DISCUSSION THREAD

**Discussion by precompute** [clojure-emacs/cider#3650] [Q&A] [answered]
Hi!  Been using Cider for a while.  This is something I have been putting up with and haven't been able to find a solution for.

Every single time I open a .clj file, cider pops up asking to connect to a nrepl (cider-connect-\*, cider-jack-in-\*).  If I don't have a nrepl active, it exits with an error.  Often, the buffer stays the same and every single time I revert I have to enter the port details manually.  I'm doing this for EVERY clojure buffer!  It's very annoying!  Is there a way to make it shut up or accept some default port settings?

Config:
```
(use-package cider
  :elpaca (:depth 1)
  :hook (clojure-mode . cider))
```

**Comment by vemv** [comment]
That's not our traditional behavior at all - if it was we'd get a lot of complaints 😄

It sounds like something is invoking `cider-mode` too early. Maybe it's some other package or bit of setup.

I'd suggest fixing that first, and then re-evaluating whether you need default connection settings - I wouldn't be too keen on facilitating workarounds.

Cheers - V

**Comment by yuhan0** [comment]
`cider` is an interactive command for jacking in / connecting to a repl, you probably meant to add `cider-mode` instead to your clojure-mode-hook. (which simply starts the minor mode and enables CIDER's bindings in the background)

**Reply by precompute** [reply]
That worked!  Thank you.
I still have to enter the details manually, though.  Is there a way around that?

**Reply by vemv** [reply, marked as answer]
https://github.com/clojure-emacs/cider/pull/3359 intended to tackle that one

However you don't particularly need it - you can make your own helper that performs `cider-connect-clj` / `cider-jack-in` with specific params.

If you normally use `cider-connect-clj`, there's not a lot to type anyway thanks to open port detection (so normally it's RET for the localhost suggestion, RET for the detected port)
